src/serialization/Portable.ts

```diff
@@ -41,127 +48,607 @@ export enum FieldType {
 }
 
 /**
  * Writer helper for {@link Portable} objects.
  */
 export interface PortableWriter {
+
+    /**
+     * Writes a number as 32-bit signed integer.
+     *
+     * @param fieldName name of the field
+     * @param value     int value to be written. The value must be a valid signed 32-bit integer.
+     * Behavior is undefined when value is anything other than a signed 32-bit integer.
+     */
     writeInt(fieldName: string, value: number): void;
 
-    writeLong(fieldName: string, long: Long): void;
+    /**
+     * Writes a long.
+     *
+     * @param fieldName name of the field
+     * @param value     long value to be written
+     */
+    writeLong(fieldName: string, value: Long): void;
 
-    /** @deprecated since version 4.2.
-     *  This method will be removed in next major version. Please use {@link writeString} instead.
+    /**
+     * Writes a string as UTF-8 encoded bytes.
+     *
+     * @param fieldName name of the field
+     * @param value     UTF string value to be written
+     * @deprecated since version 4.2 for the sake of better naming. Please use {@link writeString} instead.
+     * This method will be removed in the next major version.
      */
-    writeUTF(fieldName: string, str: string): void;
+    writeUTF(fieldName: string, value: string | null): void;
 
-    writeString(fieldName: string, str: string): void;
+    /**
+     * Writes a string as UTF-8 encoded bytes.
+     *
+     * @param fieldName name of the field
+     * @param value     UTF string value to be written
+     */
+    writeString(fieldName: string, value: string | null): void;
 
+    /**
+     * Writes a primitive boolean.
+     *
+     * @param fieldName name of the field
+     * @param value     boolean value to be written
+     */
     writeBoolean(fieldName: string, value: boolean): void;
 
+    /**
+     * Writes a number as 8-bit unsigned integer.
+     *
+     * @param fieldName name of the field
+     * @param value     Byte value to be written. Must be a valid unsigned 8-bit integer.
+     * Behaviour is undefined when value is anything other than a unsigned 8-bit integer.
+     */
     writeByte(fieldName: string, value: number): void;
 
-    writeChar(fieldName: string, char: string): void;
+    /**
+     * Writes a single character string using `char.charCodeAt(0)`.
+     *
+     * A two-byte unsigned integer representing the UTF-16
+     * code unit value of the single character string will be written.
+     *
+     * @param fieldName name of the field
+     * @param value     char value to be written
+     */
+    writeChar(fieldName: string, value: string): void;
 
-    writeDouble(fieldName: string, double: number): void;
+    /**
+     * Writes a number as double.
+     *
+     * @param fieldName name of the field
+     * @param value     double value to be written. The value must be a JavaScript number.
+     * Behavior is undefined when value is anything other than a JavaScript number.
```

Comment:
How can it be anything other than a number? Also, as far as I remember, JS number and Java double have the same range so this warning might be unnecessary

Reply:
Value has number type but in runtime anything can be passed to it. Especially in environments where typescript is not used.